snapcraft/plugins/catkin.py

```diff
@@ -1038,14 +1038,14 @@ def _run(self, arguments):
             # Source our own workspace so we have all of Catkin's dependencies,
             # then source the workspace we're actually supposed to be crawling.
             lines.append(
-                "_CATKIN_SETUP_DIR={} source {}".format(
+                "_CATKIN_SETUP_DIR={} source {} --local".format(
                     ros_path, os.path.join(ros_path, "setup.sh")
                 )
             )
 
             for workspace in self._workspaces:
                 lines.append(
-                    "_CATKIN_SETUP_DIR={} source {} --extend".format(
+                    "_CATKIN_SETUP_DIR={} source {} --local --extend".format(
```

Comment:
I haven't looked at how this is implemented, but from [the docs](https://github.com/ros/catkin/blob/2f51d6618788267192b57fbbd24ff41e8e6b4118/cmake/templates/setup.sh.in#L7-L8), `--local` and `--extend` seem mutually exclusive. Specifically, `--extend` allows chaining, building on an underlay, whereas `--local` explicitly ignores the underlay. We need the ability to chain here, thus I doubt `--local` is what we need. That said, your investigation was tremendously well done, thank you, and leads us toward another solution.

Check [this](https://github.com/snapcore/snapcraft/blob/master/snapcraft/plugins/catkin.py#L717-L725) out. We massage _setup_util.py to make sure it doesn't try to pull ROS bits off the host, but _only after we finish building_. I suggest looking into doing something similar in [_prepare_build](https://github.com/snapcore/snapcraft/blob/master/snapcraft/plugins/catkin.py#L660) as well. Is that feasible?

Reply:
You are right, `--local` will fully clear `CMAKE_PREFIX_PATH` including any underlay. Looking at your suggestion.

Reply:
After @artivis's research, it seems that `--local` actually does exactly what we need here. They aren't mutually exclusive, in fact, `--local` has the exact same effect as our `_setup_util.py` massaging, which means we can probably get rid of that hack as well. Nice find!